src/datasources/targeted-messaging/targeted-messaging.datasource.spec.ts

```diff
@@ -265,6 +438,66 @@ describe('TargetedMessagingDataSource tests', () => {
       });
     });
 
+    it('gets a submission from cache', async () => {
+      const createOutreachDto = createOutreachDtoBuilder().build();
+      const outreach = await target.createOutreach(createOutreachDto);
+      const createTargetedSafesDto = createTargetedSafesDtoBuilder()
+        .with('outreachId', outreach.id)
+        .with('addresses', [
+          getAddress(faker.finance.ethereumAddress()),
+          getAddress(faker.finance.ethereumAddress()),
+        ])
+        .build();
+      const targetedSafes = await target.createTargetedSafes(
+        createTargetedSafesDto,
+      );
+      const signerAddress = getAddress(faker.finance.ethereumAddress());
+
+      await target.createSubmission({
+        targetedSafe: targetedSafes[0],
+        signerAddress,
+      });
+
+      // first call is not cached
+      await target.getSubmission({
+        targetedSafe: targetedSafes[0],
+        signerAddress,
+      });
+      // second call is cached
+      const result = await target.getSubmission({
+        targetedSafe: targetedSafes[0],
+        signerAddress,
+      });
+
+      expect(result).toStrictEqual(
+        expect.objectContaining({
+          id: expect.any(Number),
+          outreachId: outreach.id,
+          targetedSafeId: targetedSafes[0].id,
+          signerAddress,
+          completionDate: expect.any(String),
+        }),
+      );
```

Comment:
It's because they are retrieved from the cache, and therefore parsed through `JSON.parse`  which converts the values to strings. We have this behavior in other parts of the code also. Maybe we could map back these values internally to Dates, but they will be serialised to strings on the routes level anyway. 🤔

Reply:
I think it's fine for the time being. Thank you for clarifying.